htdocs/compta/facture/card.php

```diff
@@ -270,10 +270,10 @@
 	{
 		$object->fetch($id);
 
-		// On verifie signe facture
+		// We check sign of amount
 		if ($object->type == Facture::TYPE_CREDIT_NOTE) {
-			// Si avoir, le signe doit etre negatif
-			if ($object->total_ht >= 0) {
```

Comment:
Do you have an exemple of invoice with price without tax positive and including tax negative ?

Reply:
Price without is 0, tax is negative amount, total including is same as negative tax amount:

![Customer invoice - Card 13-11-2019 19-28-31](https://user-images.githubusercontent.com/4573808/68792566-db502d00-064b-11ea-9e13-7fe7fc40bb3c.png)

The 'Import tax line' is made with some external module which creates the total vat amount.

Use case:
Guadeloupéan come in France. This customer buy product and go home
just before to take the plane the customer send to the shop the paper douane (french customs) for having the refund of tva to his invoice 
when i receive the paper douane (french customers) i need to make Avoir for just tva

Maybe there exist better solution.

For the moment I solved with some hooks on invoice card, but is quite some hook code for just one small modification.

Reply:
On you screenshot, we see that total without tax is 0, amount of vat is 500, but the details of lines, everything is zero.
So it seems your data are corrupted between lines and total of invoice.

Reply:
The line is made with some external module and contains the 500 but is not visible in the line, so example maybe not so clear. The main reason for this PR is because (not only) Dolibarr is missing possibility to add export/import VAT amounts in customer/supplier invoices to have them correctly in accountancy. It's real  VAT not error on VAT, maybe @aspangaro knows more about this. Do you think a Feature request on this could be accepted?

Reply:
When you say
"possibility to add export/import VAT amounts in customer/supplier invoices",
you mean
"possibility to add export/import VAT amounts in customer/supplier invoices **after having done the initial invoice with a zero VAT** and you need to define VAT later" ?

Reply:
Hi Eldy, i try to explain because i know the problem. 

**First situation for supplier:**
You buy products in China and you receive container to Le Havre for example. Your container go to Douane and you receive shipment invoice by SCHENKER for exemple..
So on this invoice you have VAT(brut) for the container. In my example 4000 euros (no HT for that it's VAT brut) - Cost for Douane 300 euros TTC -> 250HT and 50TVA - Cost for shipment Le Havre to your company 450euros TTC -> 375HT and 75VAT.
So you have: 
- HT : 625 euros
- VAT: 4125 euros
- TTC: 4750 euros

I work with extraprice custom for add this VAT, so i have line with HT =0 but VAT with value like the capture in this post.

**Second situation for customer:**
You have Réunion customer on holidays in France. He buy products on your shop for 300eurosTTC. When he go back to Réunion he go to Douane and need VAT refund.
So you have invoice in Dolibarr for 300TTC -> 250HT / 50VAT. Need to make Avoir for just the VAT -> 50 euros, no HT price. It's not CA just refund the VAT to the customer.

In the first case : possibility to add VAT import in the supplier invoice.
In the second case: possibility to add VAT import in customer "avoir" AFTER having done the initial invoice.

Is it clear for you ...

Reply:
We do not mind if the VAT amount entry should be done by some external module, but the main issue for the PR is the check on negative TTC (tax included) for customer VAT credit note. I need to override many Dolibarr core code to change this simple check. Should I add HIDDEN constant?

Reply:
So my understanding is both in (Case A) and (Case B) described by @ekowsol, it seems we need to be able to add line with just VAT (Amount without tax 0, VAT = x euros, Amount inc TAX = x euros).
Currently only an external module is able to do that. But what do you think of this:
We can add an option "MAIN_ALLOW_VAT_ONLY_LINES". When set:
* In the combo free line, we add "VAT only" in addition to "Product" and "Service".
* When selected, we can enter in the field price the amount of VAT so it is inserted.
* When a credit not is VAT only, we accept to validate it.
@aspangaro Do you think this is a good idea to solve the situation Case A and Case B ?